internal/orchestration/orchestration_test.go

```diff
@@ -11,176 +11,150 @@ import (
-	treePos := tree.DefaultTreePosUint32(replicas)
-	if random {
-		rnd := rand.New(rand.NewSource(int64(rand.Uint64())))
-		rnd.Shuffle(len(treePos), reflect.Swapper(treePos))
-	}
-
-	return &orchestrationpb.ReplicaOpts{
+		// Common default values:
+		ReplicaHosts:      []string{"localhost"},
+		ClientHosts:       []string{"localhost"},
+		Duration:          5 * time.Second,
 		BatchSize:         100,
-		ConnectTimeout:    durationpb.New(time.Second),
-		InitialTimeout:    durationpb.New(100 * time.Millisecond),
-		TimeoutSamples:    1000,
+		ConnectTimeout:    time.Second,
+		ViewTimeout:       100 * time.Millisecond,
+		DurationSamples:   1000,
 		TimeoutMultiplier: 1.2,
-		Consensus:         consensusImpl,
-		Crypto:            crypto,
-		LeaderRotation:    "tree-leader",
 		Modules:           mods,
-		ByzantineStrategy: "", // TODO currently kauri does not support byzantine replicas
-		BranchFactor:      uint32(bf),
-		TreePositions:     treePos,
-		TreeDelta:         durationpb.New(30 * time.Millisecond),
+		MaxConcurrent:     250,
+		PayloadSize:       100,
+		RateLimit:         math.Inf(1),
+		ClientTimeout:     500 * time.Millisecond,
+		TreeDelta:         30 * time.Millisecond,
 	}
-}
-
-func makeClientOpts() *orchestrationpb.ClientOpts {
-	return &orchestrationpb.ClientOpts{
-		ConnectTimeout: durationpb.New(time.Second),
-		MaxConcurrent:  250,
-		PayloadSize:    100,
-		RateLimit:      math.Inf(1),
-		Timeout:        durationpb.New(500 * time.Millisecond),
+	if randomTree {
+		rnd := rand.New(rand.NewSource(int64(rand.Uint64())))
+		rnd.Shuffle(len(cfg.TreePositions), reflect.Swapper(cfg.TreePositions))
 	}
+	return cfg
 }
 
-func TestOrchestration(t *testing.T) {
-	run := func(t *testing.T, replicaOpts *orchestrationpb.ReplicaOpts) {
-		t.Helper()
-
-		controllerStream, workerStream := net.Pipe()
-		workerProxy := orchestration.NewRemoteWorker(protostream.NewWriter(controllerStream), protostream.NewReader(controllerStream))
-		worker := orchestration.NewWorker(protostream.NewWriter(workerStream), protostream.NewReader(workerStream), metrics.NopLogger(), nil, 0)
-
-		cfg := config.NewLocal(7, 2)
-		cfg.TreePositions = replicaOpts.TreePositions
-		cfg.BranchFactor = replicaOpts.BranchFactor
-		cfg.TreeDelta = replicaOpts.TreeDelta.AsDuration()
-
-		experiment, err := orchestration.NewExperiment(
-			5*time.Second,
-			"",
-			replicaOpts,
-			makeClientOpts(),
-			cfg,
-			map[string]orchestration.RemoteWorker{"localhost": workerProxy},
-			logging.New("ctrl"),
-		)
-		if err != nil {
-			t.Fatal(err)
-		}
-
-		c := make(chan error)
-		go func() {
-			c <- worker.Run()
-		}()
+func run(t *testing.T, cfg *config.ExperimentConfig) {
+	t.Helper()
 
-		err = experiment.Run()
-		if err != nil {
-			t.Fatal(err)
-		}
+	controllerStream, workerStream := net.Pipe()
+	workerProxy := orchestration.NewRemoteWorker(protostream.NewWriter(controllerStream), protostream.NewReader(controllerStream))
```

Comment:
I got curious about this line and the next one; we use `controllerStream` in both on the same line, and `workerStream` in both on the next line. Does this make sense?
Looking at code from `cli/run.go`:
```go
		remoteWorkers[host] = orchestration.NewRemoteWorker(
			protostream.NewWriter(session.Stdin()), protostream.NewReader(session.Stdout()),
		)
```
We use a different stream for the reader and writer.